data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -15,11 +18,15 @@ class TransactionRepository(
     private val gatewayApi: GatewayApi
 ) {
 
-    suspend fun getTransactions(safeAddress: Solidity.Address): Page<Transaction> =
-        gatewayApi.loadTransactions(safeAddress.asEthereumAddressChecksumString())
+    suspend fun getTransactions(safeAddress: Solidity.Address): Page<UnifiedEntry> {
+        val queued = gatewayApi.loadTransactionsQueue(safeAddress.asEthereumAddressChecksumString())
+        val history = gatewayApi.loadTransactionsHistory(safeAddress.asEthereumAddressChecksumString())
 
-    suspend fun loadTransactionsPage(pageLink: String): Page<Transaction> =
-        gatewayApi.loadTransactionsPage(pageLink)
+        return Page((queued.count ?: 0) + (history.count ?: 0), history.next, history.previous, queued.results + history.results)
```

Comment:
Why are you merging both lists? I think we should have 2 separate calls for each endpoint.

Reply:
Because the UI is not prt of this PR and I still wanted to see all txs.

Reply:
Actually, if this PR is meant to provide the data layer, I would expect for all the methods in `TransactionRespository` be ready to use by the `viewModel`. 

If you need to see the aggregated result, then you could easily mix it in a unit test, as this is not a use case consumed by production code. Or if you just need it in the UI, then maybe the `viewModel` is a better place to mix this up.

Reply:
I think this is related to Vitaly's comment here: https://github.com/gnosis/safe-android/pull/1133#discussion_r542156672

To which my answer would be "yes, this should be part of the PR, IMO"